tasks/SPARK/src/main/java/kmer/Retriable.java

```diff
@@ -0,0 +1,25 @@
+package kmer;
+
+import java.util.Optional;
+import java.util.Random;
+
+public interface Retriable<T> {
+    T run() throws Exception;
+    Random random = new Random();
+
+    static <T> Optional<T> runWithRetries(int maxRetries, int maxInterval, Retriable<T> t) throws InterruptedException {
+        int count = 0;
+        while (count < maxRetries) {
+            try {
```

Comment:
Does a 503 really cause an exception?  Does this need to be tested through a unit test?

Reply:
Yep. Very Java.